backend/dataall/modules/datasets/services/dataset_service.py

```diff
@@ -435,8 +435,8 @@ def _deploy_dataset_stack(dataset: Dataset):
         stack_helper.deploy_stack(dataset.environmentUri)
 
     @staticmethod
-    def _create_dataset_stack(session, dataset: Dataset) -> Stack:
-        return Stack.create_stack(
+    def _create_dataset_stack(session, dataset: Dataset) -> StackRepository:
```

Comment:
This function should return a `Stack` model from `dataall/core/stacks/db/stack_models.py`